### ISSUE TITLE
Unable to connect lidarr to deemix indexer

### ISSUE THREAD

**Issue by lucafolin** [youegraillot/lidarr-on-steroids#63]
After a fresh installation the two GUIs work fine and I am also able to enable in lidarr deemix as a downloader, but when I try to set deemix as an indexer I get this error:

**"Unable to connect to indexer, please check your DNS settings and ensure IPv6 is working or disabled. An error occurred while sending the request."**

![Web capture_9-8-2023_75916_bb folin org](https://github.com/youegraillot/lidarr-on-steroids/assets/9470145/e4ae899f-7351-4505-9a0f-c70d91e8c160)

My docket TZ is synched with my host TZ. 
This is my log output:

```
2023-08-09 08:02:15.6|Warn|LidarrErrorPipeline|FluentValidation.ValidationException: Validation failed: 
 -- : Unable to connect to indexer, please check your DNS settings and ensure IPv6 is working or disabled. An error occurred while sending the request.

[v1.3.0.3322] FluentValidation.ValidationException: Validation failed: 
 -- : Unable to connect to indexer, please check your DNS settings and ensure IPv6 is working or disabled. An error occurred while sending the request.
   at Lidarr.Api.V1.ProviderControllerBase`3.VerifyValidationResult(ValidationResult validationResult, Boolean includeWarnings) in ./Lidarr.Api.V1/ProviderControllerBase.cs:line 208
   at Lidarr.Api.V1.ProviderControllerBase`3.Test(TProviderDefinition definition, Boolean includeWarnings) in ./Lidarr.Api.V1/ProviderControllerBase.cs:line 199
   at Lidarr.Api.V1.ProviderControllerBase`3.Test(TProviderResource providerResource) in ./Lidarr.Api.V1/ProviderControllerBase.cs:line 145
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeInnerFilterAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeNextResourceFilter>g__Awaited|25_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Rethrow(ResourceExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.InvokeFilterPipelineAsync()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Lidarr.Http.Middleware.BufferingMiddleware.InvokeAsync(HttpContext context) in ./Lidarr.Http/Middleware/BufferingMiddleware.cs:line 28
   at Lidarr.Http.Middleware.IfModifiedMiddleware.InvokeAsync(HttpContext context) in ./Lidarr.Http/Middleware/IfModifiedMiddleware.cs:line 41
   at Lidarr.Http.Middleware.CacheHeaderMiddleware.InvokeAsync(HttpContext context) in ./Lidarr.Http/Middleware/CacheHeaderMiddleware.cs:line 33
   at Lidarr.Http.Middleware.StartingUpMiddleware.InvokeAsync(HttpContext context) in ./Lidarr.Http/Middleware/StartingUpMiddleware.cs:line 38
   at Lidarr.Http.Middleware.UrlBaseMiddleware.InvokeAsync(HttpContext context) in ./Lidarr.Http/Middleware/UrlBaseMiddleware.cs:line 27
   at Lidarr.Http.Middleware.VersionMiddleware.InvokeAsync(HttpContext context) in ./Lidarr.Http/Middleware/VersionMiddleware.cs:line 29
   at Microsoft.AspNetCore.ResponseCompression.ResponseCompressionMiddleware.InvokeCore(HttpContext context)
   at Microsoft.AspNetCore.Authorization.Policy.AuthorizationMiddlewareResultHandler.HandleAsync(RequestDelegate next, HttpContext context, AuthorizationPolicy policy, PolicyAuthorizationResult authorizeResult)
   at Microsoft.AspNetCore.Authorization.AuthorizationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware.<Invoke>g__Awaited|6_0(ExceptionHandlerMiddleware middleware, HttpContext context, Task task)


2023-08-09 08:02:15.6|Warn|LidarrErrorPipeline|Invalid request Validation failed: 
 -- : Unable to connect to indexer, please check your DNS settings and ensure IPv6 is working or disabled. An error occurred while sending the request.
```
Any suggestion to fix this issue ?

**Comment by robbtate** [comment]
I also have the same issue.

If you look at the logs upon install you can see an error occur. I dont have it on hand but its something along the lines of "deemix-config/log/Lidarr.log" missing.

I have tried changing all folder permissions to allow writing to this directory and also manually creating the directory and file but this also doesnt help.

It was working fine up until a couple of days ago, what has changed? I wonder if the Lidarr image has been updated and is causing conflicts (https://github.com/Lidarr/Lidarr/issues).

**Comment by trailfog** [comment]
I have the same issue. The deemix web GUI interface works just fine and can also download music.
However, looking at the logs, it seems like the deemix server is restarting when lidarr tries to make a request to the deemix server.

This can also be confirmed, by running the `top` command and checking the deemix-servers PID before/after running a request from lidarr.

I think this is the important part in the logs:

```
2023-08-09T11:33:26.130Z [info] Listening on 0.0.0.0:6595
2023-08-09T11:34:16.233Z [info] Currently running deemix-gui version 2022.12.22-r222.5d447b6035
2023-08-09T11:34:16.237Z [info] deemix-lib version 3.6.14
2023-08-09T11:34:19.216Z [error] unhandledRejection: {"DATA_ERROR":"Page \"channels/country\" without content"}
GWAPIError: {"DATA_ERROR":"Page \"channels/country\" without content"}
    at GW.api_call (/snapshot/src/server/dist/app.js)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
GWAPIError: {"DATA_ERROR":"Page \"channels/country\" without content"}
    at GW.api_call (/snapshot/src/server/dist/app.js)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
.
. (Lidarr stack trace as in the original post)
.

[Warn] LidarrErrorPipeline: Invalid request Validation failed: 
 -- : Unable to connect to indexer, please check your DNS settings and ensure IPv6 is working or disabled. Connection refused (lidarr:6595) 
[services.d] Starting Deemix
2023-08-09T11:34:23.621Z [info] Listening on 0.0.0.0:6595
```

**Comment by manshurtigh** [comment]
I have the same problem!

**Comment by trailfog** [comment]
I have tracked down the issue to deemix-gui. Using a man in the middle proxy, the issue seems to be with deemix which tries to access the new releases page of deezer.

When visiting `http://localhost:6595/api/newReleases`, the deemix-gui server will crash and restart. Looking at the deemix-gui code, it seems like deezer has changed up their website and is requesting some additional data (`channels/country`) which is not being provided in the request.

Specifically, the following file/line crashes the deemix server: <https://gitlab.com/RemixDev/deemix-gui/-/blob/main/server/src/routes/api/get/newReleases.ts?ref_type=heads#L59>.

Wrapping that line in a simple `try {} catch(e) {}` would probably suffice for the time being, as only one of many requests to deezer fails (the one to the channel `/channels/module/39a8ce99-3d3a-4a72-b379-14b53b50b614` fails in particular).

I got mine to work by changing the above-mentioned line to:
```js
let channelData;
try {
    channelData = await dz.gw.get_page(channelName);
} catch (error) {
    console.error(`Caught error ${error}`);
    return [];
}
```

Last I read, the developer of deemix-gui is no longer actively developing the software. After such a fix, the <https://gitlab.com/bockiii/deemix-docker> repo would need to release a new build based on the updated code and then a new release of this repo is required.

TLDR: This is not actually an issue that this repo can fix directly, and there is a long chain of dependencies that needs to be updated - don't expect a fix anytime soon, unless this repo alters the source code of its base image as a temporary fix.

**Comment by seasportsnetwork** [comment]
I have been troubleshooting for almost a week now. It worked the first time I see this up, then I started getting errors with the indexer. When I tried to reconnect, it gives me the same error as you. I have done fresh installs multiple times and nothing has worked.

**Comment by seasportsnetwork** [comment]
> I have tracked down the issue to deemix-gui. Using a man in the middle proxy, the issue seems to be with deemix which tries to access the new releases page of deezer.
> 
> When visiting `http://localhost:6595/api/newReleases`, the deemix-gui server will crash and restart. Looking at the deemix-gui code, it seems like deezer has changed up their website and is requesting some additional data (`channels/country`) which is not being provided in the request.
> 
> Specifically, the following file/line crashes the deemix server: https://gitlab.com/RemixDev/deemix-gui/-/blob/main/server/src/routes/api/get/newReleases.ts?ref_type=heads#L59.
> 
> Wrapping that line in a simple `try {} catch(e) {}` would probably suffice for the time being, as only one of many requests to deezer fails (the one to the channel `/channels/module/39a8ce99-3d3a-4a72-b379-14b53b50b614` fails in particular).
> 
> I got mine to work by changing the above-mentioned line to:
> 
> ```js
> let channelData;
> try {
>     channelData = await dz.gw.get_page(channelName);
> } catch (error) {
>     console.error(`Caught error ${error}`);
>     return [];
> }
> ```
> 
> Last I read, the developer of deemix-gui is no longer actively developing the software. After such a fix, the https://gitlab.com/bockiii/deemix-docker repo would need to release a new build based on the updated code and then a new release of this repo is required.
> 
> TLDR: This is not actually an issue that this repo can fix directly, and there is a long chain of dependencies that needs to be updated - don't expect a fix anytime soon, unless this repo alters the source code of its base image as a temporary fix.

did you make this change in your deemix config? are you able to share?

**Comment by trailfog** [comment]
@seasportsnetwork I have just opened a pull request with a temporary fix. We will have to now wait until it gets merged, or the issue gets fixed in another way.

This is not a configuration issue, but one with the source code of deemix(-gui).

**Comment by trailfog** [comment]
@kaibagley I created a pull request with a superficial fix where you can see how one could patch the issue (See #65).

It's probably best if you just wait for the pull request to be merged by @youegraillot and a new image of `lidarr-on-steroids` to be published.

Alternatively, if you are feeling daring and would like to build your own image with the fix for the time being, you could run the following commands:

``` bash
git clone https://github.com/trailfog/lidarr-on-steroids.git && cd lidarr-on-steroids
docker build -t my-lidarr-on-steroids . && docker run \
   -p 6595:6595 -p 8686:8686 \
   --name my-lidarr-on-steroids-container \
   -v <path>:/config \
   -v <path>:/config_deemix \
   -v <path>:/downloads \
   -v <path>:/music \
   --restart unless-stopped \
   my-lidarr-on-steroids
```

**Comment by seasportsnetwork** [comment]
> @seasportsnetwork I have just opened a pull request with a temporary fix. We will have to now wait until it gets merged, or the issue gets fixed in another way.
> 
> 
> 
> This is not a configuration issue, but one with the source code of deemix(-gui).

I have only started using docker.. so I'm learning. How do I deploy the container with your fix? Thanks for your help!

**Comment by trailfog** [comment]
@seasportsnetwork See the comment above to kaibagley.
Unless you are very desperate, I would just wait for the fix to be merged instead of building your own image.

**Comment by robbtate** [comment]
I've gave up on this sinking ship and moved to unmanic. It works great, download artists/albums/songs/playlist straight from deemix and unmanic transcodes from flac to MP3 in the same folder. No faffing around it just works.

**Comment by seasportsnetwork** [comment]
> transcodes

I'm gonna check it out!

**Comment by mon5termatt** [comment]
> I've gave up on this sinking ship and moved to unmanic. It works great, download artists/albums/songs/playlist straight from deemix and unmanic transcodes from flac to MP3 in the same folder. No faffing around it just works.

This is unhelpful, it doesnt automatically download.

**Comment by janaxhell** [comment]
> When visiting `http://localhost:6595/api/newReleases`, the deemix-gui server will crash and restart. Looking at the deemix-gui code, it seems like deezer has changed up their website and is requesting some additional data (`channels/country`) which is not being provided in the request.

Besides confirming that I have the same issue on Docker, I don't on Windows: Deemix for Windows works fine as always. Just to narrow down the variables. I'm on
```
Current WebUI Version: 1.9.3
Current Version: 2022.12.14-r222.5d447b6035
deemix lib version: 3.6.14
```

**Comment by mbc0** [comment]
I also have this issue on a fresh install on a Dedicated Server but working fine on my unraid server

**Comment by moutonnoireu** [comment]
For people who don't want to wait, I just published an image based on @trailfog modifications, it's here : moutonnoir/lidarr-on-steroids. I just applied the fix, I deserve no credits. 👍

**Comment by noworthwhile** [comment]
Any idea how I can apply the fix on the Windows version of the Deemix app?

**Comment by janaxhell** [comment]
> Any idea how I can apply the fix on the Windows version of the Deemix app?

This patch is for Lidarr-on-Steroids, not Deemix. It's to re-enable the usage of Deemix from Lidarr. Also Deemix for Windows works fine.

**Comment by noworthwhile** [comment]
> > Any idea how I can apply the fix on the Windows version of the Deemix app?
> 
> This patch is for Lidarr-on-Steroids, not Deemix. It's to re-enable the usage of Deemix from Lidarr. Also Deemix for Windows works fine.

Same issue is present when using the plugin branch of Lidarr for Windows, connected to the Deemix-GUI windows app. 
Both pieces of software work independently but not together.
I'm assuming that the fix would work for this instance.

**Comment by noworthwhile** [comment]
> For people who don't want to wait, I just published an image based on @trailfog modifications, it's here : [moutonnoir/lidarr-on-steroids](https://hub.docker.com/r/moutonnoir/lidarr-on-steroids/tags). I just applied the fix, I deserve no credits. 👍

Any way to make one for arm architecture? (Raspberrypi) ? 

My knowledge of building dockerfiles is really poor

**Comment by Cameri** [comment]
@moutonnoireu thanks, your image worked!

**Comment by Eisteed** [comment]
> For people who don't want to wait, I just published an image based on @trailfog modifications, it's here : [moutonnoir/lidarr-on-steroids](https://hub.docker.com/r/moutonnoir/lidarr-on-steroids/tags). I just applied the fix, I deserve no credits. 👍

Thanks a lot for releasing this, it does fix the issue for now !

**Comment by estebanthi** [comment]
> @kaibagley I created a pull request with a superficial fix where you can see how one could patch the issue (See #65).
> 
> It's probably best if you just wait for the pull request to be merged by @youegraillot and a new image of `lidarr-on-steroids` to be published.
> 
> Alternatively, if you are feeling daring and would like to build your own image with the fix for the time being, you could run something along the following commands:
> 
> ```shell
> git clone https://github.com/trailfog/lidarr-on-steroids.git && cd lidarr-on-steroids
> docker build -t my-lidarr-on-steroids . && docker run \
>    -p 6595:6595 -p 8686:8686 \
>    --name my-lidarr-on-steroids-container \
>    -v <path>:/config \
>    -v <path>:/config_deemix \
>    -v <path>:/downloads \
>    -v <path>:/music \
>    --restart unless-stopped \
>    my-lidarr-on-steroids
> ```

Hey, I'm trying to build the image for Raspberry Pi, but I keep getting an error because [cr.hotio.dev/hotio/lidarr:pr-plugins-1.3.0.3322](https://cr.hotio.dev/v2/hotio/lidarr/manifests/pr-plugins-1.3.0.3322) seems to require authentication. How did you manage to build it?

```
 => ERROR [internal] load metadata for cr.hotio.dev/hotio/lidarr:pr-plugins-1.3.0.3322                                                                                                                                                   0.1s
------
 > [internal] load metadata for cr.hotio.dev/hotio/lidarr:pr-plugins-1.3.0.3322:
------
Dockerfile:3
--------------------
   1 |     FROM registry.gitlab.com/bockiii/deemix-docker:latest@sha256:d325660d833c30c8bbb5cf0caa75135b42a2f69b82f2269fd5bea91a04c449e6 as deemix
   2 |     
   3 | >>> FROM cr.hotio.dev/hotio/lidarr:pr-plugins-1.3.0.3322
   4 |     
   5 |     LABEL maintainer="youegraillot"
--------------------
ERROR: failed to solve: cr.hotio.dev/hotio/lidarr:pr-plugins-1.3.0.3322: failed to do request: Head "https://cr.hotio.dev/v2/hotio/lidarr/manifests/pr-plugins-1.3.0.3322": dial tcp 0.0.0.0:443: connect: connection refused
```

**Comment by trailfog** [comment]
> > @kaibagley I created a pull request with a superficial fix where you can see how one could patch the issue (See #65).
> > It's probably best if you just wait for the pull request to be merged by @youegraillot and a new image of `lidarr-on-steroids` to be published.
> > Alternatively, if you are feeling daring and would like to build your own image with the fix for the time being, you could run something along the following commands:
> > ```shell
> > git clone https://github.com/trailfog/lidarr-on-steroids.git && cd lidarr-on-steroids
> > docker build -t my-lidarr-on-steroids . && docker run \
> >    -p 6595:6595 -p 8686:8686 \
> >    --name my-lidarr-on-steroids-container \
> >    -v <path>:/config \
> >    -v <path>:/config_deemix \
> >    -v <path>:/downloads \
> >    -v <path>:/music \
> >    --restart unless-stopped \
> >    my-lidarr-on-steroids
> > ```
> 
> Hey, I'm trying to build the image for Raspberry Pi, but I keep getting an error because [cr.hotio.dev/hotio/lidarr:pr-plugins-1.3.0.3322](https://cr.hotio.dev/v2/hotio/lidarr/manifests/pr-plugins-1.3.0.3322) seems to require authentication. How did you manage to build it?
> 
> ```
>  => ERROR [internal] load metadata for cr.hotio.dev/hotio/lidarr:pr-plugins-1.3.0.3322                                                                                                                                                   0.1s
> ------
>  > [internal] load metadata for cr.hotio.dev/hotio/lidarr:pr-plugins-1.3.0.3322:
> ------
> Dockerfile:3
> --------------------
>    1 |     FROM registry.gitlab.com/bockiii/deemix-docker:latest@sha256:d325660d833c30c8bbb5cf0caa75135b42a2f69b82f2269fd5bea91a04c449e6 as deemix
>    2 |     
>    3 | >>> FROM cr.hotio.dev/hotio/lidarr:pr-plugins-1.3.0.3322
>    4 |     
>    5 |     LABEL maintainer="youegraillot"
> --------------------
> ERROR: failed to solve: cr.hotio.dev/hotio/lidarr:pr-plugins-1.3.0.3322: failed to do request: Head "https://cr.hotio.dev/v2/hotio/lidarr/manifests/pr-plugins-1.3.0.3322": dial tcp 0.0.0.0:443: connect: connection refused
> ```

Seems like that image no longer exists. I haven't tested the new image I have suggested below, but probably you should be fine if you change the

```
FROM cr.hotio.dev/hotio/lidarr:pr-plugins-1.3.0.3322
```

line to the following in the Dockerfile:

```
FROM ghcr.io/hotio/lidarr:pr-plugins
```

This image should also work with RPI (amd64/arm64 versions available)

**Comment by estebanthi** [comment]
Thanks, with a few adjustments to the Dockerfile I managed to build an image, however it's not working because I can't run `/deemix-server` from inside the container:

```
d44218554c2b:/# /deemix-server
bash: /deemix-server: cannot execute: required file not found
```

I tried `ldd` to find out what's wrong:
```
d44218554c2b:/# ldd deemix-server 
        /lib/ld-linux-aarch64.so.1 (0x7fb1e64000)
        libdl.so.2 => /lib/ld-linux-aarch64.so.1 (0x7fb1e64000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x7fb1c20000)
        libm.so.6 => /lib/ld-linux-aarch64.so.1 (0x7fb1e64000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x7fb1bef000)
        libpthread.so.0 => /lib/ld-linux-aarch64.so.1 (0x7fb1e64000)
        libc.so.6 => /lib/ld-linux-aarch64.so.1 (0x7fb1e64000)
Error loading shared library ld-linux-aarch64.so.1: No such file or directory (needed by deemix-server)
Error relocating deemix-server: __getauxval: symbol not found
Error relocating deemix-server: __register_atfork: symbol not found
Error relocating deemix-server: makecontext: symbol not found
Error relocating deemix-server: backtrace: symbol not found
Error relocating deemix-server: setcontext: symbol not found
Error relocating deemix-server: getcontext: symbol not found
Error relocating deemix-server: gnu_get_libc_version: symbol not found
Error relocating deemix-server: __strdup: symbol not found
Error relocating deemix-server: __libc_stack_end: symbol not found
```

Not sure how to solve this, I'll just wait for your PR to be merged and the updated image to be released.

**Comment by youegraillot** [comment]
#65 has been merged
closing
